src/server/scripts/Outland/BlackTemple/boss_shade_of_akama.cpp

```diff
@@ -25,251 +25,175 @@ enum Says
     SPELL_AKAMA_SOUL_CHANNEL        = 40447,
+    SPELL_FIXATE                    = 40607,
+    SPELL_AKAMA_SOUL_RETRIEVE       = 40902,    // epilogue
+    SPELL_AKAMA_SOUL_EXPEL_CHANNEL  = 40927,    // epilogue
+
+    // Shade & Channelers
     SPELL_SHADE_SOUL_CHANNEL        = 40401,
-    SPELL_CHAIN_LIGHTNING           = 39945,
-    SPELL_DESTRUCTIVE_POISON        = 40874,
+    SPELL_THREAT                    = 41602,
     SPELL_SHADE_OF_AKAMA_TRIGGER    = 40955,
-    SPELL_AKAMA_SOUL_RETRIEVE       = 40902,
 
-    SPELL_ASHTONGUE_WAVE_B           = 42035,
-    SPELL_SUMMON_ASHTONGUE_SORCERER  = 40476,
-    SPELL_SUMMON_ASHTONGUE_DEFENDER  = 40474
+    // Summons
+    SPELL_ASHTONGUE_WAVE_A          = 42073,   // unused
+    SPELL_ASHTONGUE_WAVE_B          = 42035,
+    SPELL_SUMMON_ASHTONGUE_SORCERER = 40476,
+    SPELL_SUMMON_ASHTONGUE_DEFENDER = 40474
 };
 
 enum Creatures
 {
-    NPC_ASHTONGUE_CHANNELER         = 23421,
-    NPC_CREATURE_GENERATOR_AKAMA    = 23210,
     NPC_ASHTONGUE_SORCERER          = 23215,
+    NPC_ASHTONGUE_DEFENDER          = 23216,
+    NPC_ASHTONGUE_ELEMENTAL         = 23523,
+    NPC_ASHTONGUE_ROGUE             = 23318,
+    NPC_ASHTONGUE_SPIRITBIND        = 23524,
     NPC_ASHTONGUE_BROKEN            = 23319
 };
 
 enum Misc
 {
     SUMMON_GROUP_BROKENS            = 1,
 
-    POINT_START                     = 0,
-    POINT_CHANNEL_SOUL              = 1,
-
-    ACTION_AKAMA_DIED               = 1,
-    ACTION_START_ENCOUNTER          = 2,
-    ACTION_STOP_SPAWNING            = 3,
-    ACTION_DESPAWN_ALL              = 4,
-    ACTION_CHANNELERS_START_CHANNEL = 5,
-    ACTION_KILL_CHANNELERS          = 6,
-    ACTION_NO_SORCERERS             = 7,
-    ACTION_SHADE_DIED               = 8,
-
-    EVENT_AKAMA_START_ENCOUNTER     = 1,
-    EVENT_AKAMA_START_CHANNEL       = 2,
-    EVENT_SPELL_CHAIN_LIGHTNING     = 4,
-    EVENT_SPELL_DESTRUCTIVE_POISON  = 5,
-
-    EVENT_SHADE_CHECK_DISTANCE      = 10,
-    EVENT_SHADE_RESET_ENCOUNTER     = 11,
-    EVENT_SHADE_GATHER_NPCS         = 12,
-
-    EVENT_SUMMON_WAVE_B             = 20,
-    EVENT_SUMMON_ASHTONGUE_SORCERER = 21,
-    EVENT_SUMMON_ASHTONGUE_DEFENDER = 22,
-
-    EVENT_AKAMA_SCENE0              = 29,
-    EVENT_AKAMA_SCENE1              = 30,
-    EVENT_AKAMA_SCENE2              = 31,
-    EVENT_AKAMA_SCENE3              = 32,
-    EVENT_AKAMA_SCENE4              = 33,
-    EVENT_AKAMA_SCENE5              = 34,
-    EVENT_AKAMA_SCENE6              = 35,
-    EVENT_AKAMA_SCENE7              = 36
+    POINT_ENGAGE                    = 0,
+    POINT_OUTRO                     = 1,
+
+    ACTION_GENERATOR_START          = 1,
+    ACTION_GENERATOR_STOP           = 2,
+    ACTION_GENERATOR_DESPAWN_ALL    = 3,
+
+    COUNTER_SPAWNS_MAX              = 20,   // Max number of spawns for each generator, number chosen at random
+
+    ACTION_AKAMA_START_OUTRO        = 1,
+
+    FACTION_DEFAULT                 = 1820,
+    FACTION_ENGAGE                  = 1868
```

Comment:
1866？

Reply:
1868 is sniffed. Shade of Akama itself has a Monster Spar faction
```
ServerToClient: SMSG_UPDATE_OBJECT (0x27C3) Length: 92 ConnIdx: 1 Time: 11/25/2022 17:57:10.681 Number: 52899
NumObjUpdates: 1
MapID: 564 (564)
HasRemovedObjects: False
Data size: 81
[0] UpdateType: Values
[0] Object Guid: Full: \\\\ Creature/0 R3881/S31776 Map: 564 Entry: 23191 Low: 75740
[0] [0] ChannelObjects: Full: \\\\ Creature/0 R3881/S31776 Map: 564 Entry: 22841 Low: 75740
[0] (ChannelData) SpellID: 40447
[0] (ChannelData) (SpellVisual) SpellXSpellVisualID: 18249
[0] (ChannelData) (SpellVisual) ScriptVisualID: 0
[0] FactionTemplate: 1868
[0] Flags: 262144
[0] [0] NpcFlags: 0
```